tests/test_zoned_datetime.py

```diff
@@ -900,8 +922,12 @@ def test_disambiguate(self):
 
     def test_nonexistent(self):
         d = ZonedDateTime(2023, 3, 26, 1, 15, 30, tz="Europe/Amsterdam")
-        with pytest.raises(DoesntExistInZone):
+        with pytest.raises(DoesntExistInZone) as e:
```

Comment:
`pytest.raises` has a nice `match=` argument that is more idiomatic.
It even allows regex, so you can be a bit more flexible in the assertion.

Of course, you can also assert the entire message exactly—both are fine.

```python
with pytest.raises(DoesntExistInZone, match="2:15:30.*Europe/Amsterdam"):
    ...
```

Reply:
That would clean up the tests! Nice.
I will assert the entire message though, since there is some freedom in how the tested code constructs the exception.